impl/src/main/java/org/jboss/weld/bootstrap/FastAnnotatedTypeLoader.java

```diff
@@ -127,13 +128,19 @@
         }
         return null;
     }
-
-    private boolean isJandexSetFlagsAvailable() {
-        try {
-            Method setFlags = AccessController.doPrivileged(GetDeclaredMethodAction.of(ClassFileInfo.class, "setFlags", short.class));
-            return setFlags != null;
-        } catch (Exception e) {
-            return false;
+    // checking availability of ClassInfo.setFlags method is just workaround for JANDEX-37
+    private boolean initCheckTypeModifiers() {
+        if (Reflections.isClassLoadable(CLASSINFO_CLASS_NAME, new ClassLoaderResourceLoader(classFileServices.getClass().getClassLoader()))) {
```

Comment:
I would first try to load the class and if it's not null then try to find the method - right now, `ClassLoaderResourceLoader` is created and `ResourceLoader.classForName(String)}}` is invoked twice.